ext/openssl/openssl.c

```diff
@@ -3566,13 +3631,26 @@ static EVP_PKEY *php_openssl_pkey_from_zval(zval *val, int public_key, char *pas
 				TMP_CLEAN;
 			}
 		}
+		if (Z_STRLEN_P(val) > 7 && memcmp(Z_STRVAL_P(val), "pkcs11:", sizeof("pkcs11:") - 1) == 0) {
+			engine = php_openssl_make_pkcs11_engine(true);
+			if (engine == NULL) {
+				TMP_CLEAN;
+			}
+		}
 		/* it's an X509 file/cert of some kind, and we need to extract the data from that */
 		if (public_key) {
-			cert = php_openssl_x509_from_str(Z_STR_P(val));
+			if (engine) {
+				key = ENGINE_load_public_key(engine, Z_STRVAL_P(val), NULL, NULL);
+				ENGINE_finish(engine);
+				engine = NULL;
+			}
+			/* val could be a certificate (file, pkcs11:, etc., let's try to extract the key */
```

Comment:
NIT: missing `)` in the comment :)

Reply:
done https://github.com/php/php-src/pull/6860/commits/27d8f44af90e66dcdb423979a15dc91f4d7f83e4